libs/ui/src/hooks/use_devices.ts

```diff
@@ -38,6 +39,7 @@ export interface Devices {
   precinctScanner?: KioskBrowser.Device;
   batchScanner?: KioskBrowser.Device;
   printer?: KioskBrowser.PrinterInfo;
+  patHidInput?: KioskBrowser.Device;
```

Comment:
This moves us a bit further away from removing our reliance on kiosk-browser. Is there an easy way to read the PAT device status from the backend?

Reply:
I spent some time experimenting with this. We can successfully read PAT device status and input from the backend using `node-hid` but I think the frontend still needs access to the device because its primary purpose outside of this calibration flow is as an input device during the voting flow. We could build the calibration flow in the backend without much trouble because there are only 4 states. But it feels like listening to PAT input events and transmitting them to the frontend to handle DOM navigation would be messy. eg. the frontend would need to subscribe to or poll the backend for every single PAT event and simulate a browser `Tab` keypress/click event in response.

As long as we need the frontend to be aware of the PAT device for general voting flow, the marginal cost of using the frontend-integrated PAT device for the calibration flow is small to none.

That said, I'm not up to date on what the plan is to get rid of kiosk-browser. Is there something you had in mind that would be a cleaner solution than the one I mentioned above?

Reply:
It seems like there are two different aspects to interacting with the PAT device:
- Device status (is it connected?)
- Device input (select/navigate triggers)

I think we currently use custom kiosk-browser functionality for device status (via this `useDevices` hook), but use regular web browser standard interfaces for device input. Is that true?

Eventually I think we want to move device status to the backend so we can migrate away from kiosk-browser. I was suggesting that this might be a good time to start moving device status to the backend. I totally agree that it makes sense to keep using the web browser standard interface for input.

But, I also think that continuing to use the existing device status from kiosk browser doesn't make it that much harder to move the backend later, so it's also fine to leave it as is for now.

Reply:
I was thinking device management should be all-or-nothing in the frontend or backend, but that might apply less to input devices and more to devices like the card reader. I agree that we can break down the concerns cleanly between status and input. I'll update this branch to remove kiosk-browser status and move it to the backend.